.ci/Dockerfile

```diff
@@ -1,5 +1,5 @@
 # Docker image used for continuous integration
-FROM docker.io/library/golang:1.19.5
+FROM docker.io/library/golang:1.20.2
```

Comment:
Looks like it has been reverted on purpose: https://github.com/elastic/cloud-on-k8s/pull/6420 
I tried to run `make lint` on a locally built CI image but it's failing with:

```
GOGC=50 golangci-lint run --verbose
INFO [config_reader] Config search paths: [./ /go/src/github.com/elastic/cloud-on-k8s /go/src/github.com/elastic /go/src/github.com /go/src /go / /root]
INFO [config_reader] Used config file .golangci.yml
INFO [lintersdb] Active 40 linters: [asciicheck bodyclose deadcode dogsled durationcheck errcheck errorlint exhaustive exportloopref forcetypeassert gocritic goheader goimports goprintffuncname gosec gosimple govet importas ineffassign makezero misspell nakedret nestif nilerr noctx nolintlint prealloc predeclared revive staticcheck structcheck thelper tparallel typecheck unconvert unparam unused varcheck wastedassign whitespace]
INFO [loader] Go packages loading at mode 575 (name|types_sizes|compiled_files|deps|exports_file|files|imports) took 3m16.073783471s
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 174.117366ms
INFO [linters context] importas settings found, but no aliases listed. List aliases under alias: key.
make: *** [Makefile:184: lint] Killed
```

Not sure what's going on, should we create an issue and add a comment to explain to the reviewers why we should not update the image here?

Reply:
I should have at least left a comment. The goal was and still is to get rid of this image (Jenkins ci docker image).
We should soon be able to delete it and replace it with the bk agent image to avoid maintaining two images that are almost similar.